discovery/azure/azure.go

```diff
@@ -406,18 +404,19 @@ func (client *azureClient) getVMs() ([]virtualMachine, error) {
 
 func (client *azureClient) getScaleSets() ([]compute.VirtualMachineScaleSet, error) {
 	var scaleSets []compute.VirtualMachineScaleSet
-	result, err := client.vmss.ListAll()
+	ctx := context.Background()
+	result, err := client.vmss.ListAll(ctx)
 	if err != nil {
 		return scaleSets, fmt.Errorf("could not list virtual machine scale sets: %s", err)
 	}
-	scaleSets = append(scaleSets, *result.Value...)
+	scaleSets = append(scaleSets, result.Values()...)
 
-	for result.NextLink != nil {
-		result, err = client.vmss.ListAllNextResults(result)
+	for result.NotDone() {
+		err = result.NextWithContext(ctx)
 		if err != nil {
 			return scaleSets, fmt.Errorf("could not list virtual machine scale sets: %s", err)
```

Comment:
You should return nil here, not scaleSets. And above.

Reply:
Done!